libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -457,10 +457,7 @@ void AP_Airspeed::update(bool log)
     check_sensor_failures();
 
     if (log) {
-        AP_Logger *logger = AP_Logger::get_singleton();
-        if (logger != nullptr) {
-            logger->Write_Airspeed(*this);
-        }
+        AP::logger().Write_Airspeed();
```

Comment:
A bit more of a change, but what if we moved the entire body of `Write_Airspeed` in here? The sensor is already doing the logging call for us anyways, and I actually like seeing it move up to the sensor level. It also has the follow up benefit of several methods can be removed/made private, which reduces the chance of someone polling AP_Airspeed for data they shouldn't be using. (IE `corrected_pressure`)

Reply:
Pushed up patches for that.